numba_rvsdg/rendering/rendering.py

```diff
@@ -95,54 +92,160 @@ def render_block(self, digraph: "Digraph", name: str, block: BasicBlock):
-        for name, block in blocks.items():
-            for dst in block.jump_targets:
-                if dst in blocks:
-                    if type(block) in (
-                        PythonBytecodeBlock,
-                        BasicBlock,
-                        SyntheticBlock,
-                        SyntheticAssignment,
-                        SyntheticExitingLatch,
-                        SyntheticExitBranch,
-                        SyntheticHead,
-                        SyntheticExit,
-                        SyntheticBranch,
-                    ):
-                        self.g.edge(str(name), str(dst))
-                    elif type(block) == RegionBlock:
-                        if block.exiting is not None:
-                            self.g.edge(str(block.exiting), str(dst))
-                        else:
-                            self.g.edge(str(name), str(dst))
-                    else:
-                        raise Exception("unreachable " + str(block))
-            for dst in block.backedges:
-                # assert dst in blocks
-                self.g.edge(
-                    str(name), str(dst), style="dashed", color="grey", constraint="0"
-                )
+    def render_edges(self, scfg: SCFG):
+        blocks = dict(scfg)
+        def find_base_header(block: BasicBlock):
+            if isinstance(block, RegionBlock):
+                block = blocks[block.header]
+                block = find_base_header(block)
+            return block
+
+        for _, src_block in blocks.items():
+            if isinstance(src_block, RegionBlock):
+                continue
+            src_block = find_base_header(src_block)
+            for dst_name in src_block.jump_targets:
+                dst_name = find_base_header(blocks[dst_name]).name
+                if dst_name in blocks.keys():
+                    self.g.edge(str(src_block.name), str(dst_name))
+                else:
+                    raise Exception("unreachable " + str(src_block))
+            for dst_name in src_block.backedges:
+                dst_name = find_base_header(blocks[dst_name]).name
+                if dst_name in blocks.keys():
+                    self.g.edge(
+                        str(src_block.name), str(dst_name), style="dashed", color="grey", constraint="0"
+                    )
+                else:
+                    raise Exception("unreachable " + str(src_block))
 
     def render_byteflow(self, byteflow: ByteFlow):
         self.bcmap_from_bytecode(byteflow.bc)
-
         # render nodes
         for name, block in byteflow.scfg.graph.items():
             self.render_block(self.g, name, block)
-        self.render_edges(byteflow.scfg.graph)
+        self.render_edges(byteflow.scfg)
         return self.g
 
-    def render_scfg(self, scfg):
+    def bcmap_from_bytecode(self, bc: dis.Bytecode):
+        self.bcmap: Dict[int, dis.Instruction] = SCFG.bcmap_from_bytecode(bc)
+
+
+class SCFGRenderer:
```

Comment:
There seems quite a lot of duplication here. Would it be possible to share more code with the ByteFlowRenderer.

Reply:
Fixed in 96dd9fc